**1. Dalam suatu laporan oleh peneliti, dinyatakan bahwa tikus yang semula
mempunyai jangka hidup rata-rata 32 bulan dapat diperpanjang menjadi 40
bulan bila 40% kalori dalam makanannya diganti dengan protein dan vitamin.
Apakah ada alasan untuk mempercayai bahwa µ < 40 bila 64 ekor tikus
dicobakan dalam penelitian ini agar mencapai jangka hidup rata-rata 38 bulan
dengan simpang baku 5,8 bulan? Gunakan taraf nyata 0,025**

In [14]:
from scipy.stats import t

n = 64
rata_sampel = 38
simpangan_baku = 5.8
rata_populasi = 40
alpha = 0.025

stat = (rata_sampel - rata_populasi) / (simpangan_baku / (n ** 0.5))
critical = t.ppf(alpha, df = n - 1)
value = t.cdf(stat, df = n - 1)

print("Value t-statistik:", stat)
print("Value t-kritis:", critical)
print("Value:", value)

Value t-statistik: -2.7586206896551726
Value t-kritis: -1.9983405425207414
Value: 0.003794308025801665


hipotesis tersebut ditolak, karena,
- -2.76 tidak berada di rentang -1.99 dan 1.99
- p_value lebih kecil dari alpha

**2.  Ujilah hipotesis bahwa isi kaleng rata-rata suatu jenis minyak pelumas adalah 10 liter. Apabila isi suatu contoh acak 10 kaleng adalah 10.2, 9.7, 10.1, 10.3, 10.1, 9.8, 9.9, 10.4, 10.3, dan 9.8 liter. Gunakan taraf nyata 0.01 dan asumsikan bahwa isi terrsebut menyebar normal.**

In [16]:
from scipy.stats import t
import numpy as np

n = 10
rpopulasi = 10
alpha = 0.01
sample = [10.2, 9.7, 10.1, 10.3, 10.1, 9.8, 9.9, 10.4, 10.3, 9.8]
rsampel = np.mean(sample)
simpangan_baku = np.std(sample, ddof = 1)

stat = (rsampel - rpopulasi) / (simpangan_baku / (n ** 0.5))
critical = t.ppf(alpha, df = n - 1)
value = t.cdf(stat, df = n - 1)

print("Value t-statistik:", stat)
print("Value t-kritis:", critical)
print("Value:", value)


Value t-statistik: 0.771743633141273
Value t-kritis: -2.8214379250258084
Value: 0.7699755088645168


hipotesis tersebut diterima, karena,
- 0.77 berada di rentang -2.82 dan 2.82
- p_value lebih besar dari alpha

**3. Menurut penelitian konsumsi sodium yang tinggi mungkin berhubungan
dengan sakit bisul, kanker perut, dan sakit kepala. Manusia membutuhkan
garam hanya 220 milligram per hari dan ini sudah dilampaui oleh kandungan
satu porsi sereal siap makan.
Apabila suatu contoh acak 20 porsi sereal mempunyai kandungan sodium
rata-rata 244 milligram dengan simpangan baku 24,5 milligram apakah ini
menunjukkan pada taraf nyata 0,05 bahwa kandungan sodium rata-rata satu
porsi sereal lebih daripada 220 milligram?
Asumsikan bahwa sebaran kandungan sodium adalah normal**

In [18]:
from scipy.stats import t

n = 20
rata2sampel = 244
simpangan_baku = 24.5
alpha = 0.05
rata2populasi = 220

tstat = (rata2sampel - rata2populasi) / (simpangan_baku / (n ** 0.5))
tcritical = t.ppf(alpha, df = n - 1)
tvalue = t.cdf(tstat, df = n - 1)

print("Value t-statistik:", tstat)
print("Value t-kritis:", tcritical)
print("Value:", tvalue)

Value t-statistik: 4.380867874285302
Value t-kritis: -1.7291328115213678
Value: 0.999839296938128


hipotesis tersebut ditolak, karena,
- 4.38 tidak berada di rentang -1.73 dan 1.73
- p_value lebih besar dari alpha

**4.  Sebuah perusahaan menyatakan bahwa kekuatan rentangan rata-rata tali A
melebihi kekuatan rentangan tali B sebesar sekurang-kurangnya 12 kg. Untuk
diuji pernyataan ini, 50 tali dari masing-masing jenis diuji dibawah kondisi
yang sama.
Hasil uji menunjukkan tali A mempunyai kekuatan rentangan rata-rata 86,7 kg
dengan simpangan baku 6,28 kg, sedangkan tali B mempunyai kekuatan
rentangan rata-rata 77,8 kg dengan simpangan baku 5,61 kg.
Ujilah pernyataan perusahaan tersebut dengan menggunakan taraf nyata
0,05**

In [20]:
import numpy as np
from scipy.stats import t

nA = 50
nB = 50
rataA = 86.7
rataB = 77.8
stdA = 6.28
stdB = 5.61
selisih = 12
alpha = 0.05

cstat = (rataA - rataB - selisih) / np.sqrt((stdA ** 2 / nA) + (stdB ** 2 / nB))
ccritical = t.ppf(alpha, df = nA + nB - 2)
pvalue = t.cdf(cstat, df = nA + nB - 2)

print("Value z-statistik:", cstat)
print("Value z-kritis:", ccritical)
print("Value:", pvalue)

Value z-statistik: -2.603103416514782
Value z-kritis: -1.6605512170440575
Value: 0.005336424215676501


hipotesis tersebut ditolak, karena,
- -2.60 tidak berada di rentang -1.66 dan 1.66
- p_value lebih kecil dari alpha